render/process.go

```diff
@@ -124,6 +121,32 @@ func (e endpoints2Processes) Render(rpt report.Report) Nodes {
 	return ret.result()
 }
 
+// When there is more than one connection originating from a source
+// endpoint, we cannot be sure that its pid is associated with all of
+// them, since the source endpoint may have been re-used by a
+// different process. See #2665. It is better to drop such an endpoint
+// than risk rendering bogus connections.  Aliased connections - when
+// all the remote endpoints represent the same logical endpoint, due
+// to NATing - are fine though.
+func hasMoreThanOneConnection(n report.Node, endpoints report.Nodes) bool {
+	if len(n.Adjacency) < 2 {
+		return false
+	}
+	realEndpointID := ""
+	for _, endpointID := range n.Adjacency {
+		if ep, ok := endpoints[endpointID]; ok {
+			if copyID, _, ok := ep.Latest.LookupEntry(endpoint.CopyOf); ok {
+				endpointID = copyID
+			}
+		}
+		if realEndpointID != "" && realEndpointID != endpointID {
```

Comment:
convince me that this is the correct test for 3 or 4 items in `n.Adjacency`

Reply:
See above. The loop exists (by the `return false`) when encountering a 2nd real endpoint.